company-box.el

```diff
@@ -739,39 +890,44 @@ It doesn't nothing if a font icon is used."
 ;; ;;          selection n-elements percent percent-display height height-scrollbar height-blank height (+ height-scrollbar height-blank))
 ;; ;; (message "HEIGHT-S-1: %s HEIGHT-B-1: %s sum: %s" scrollbar-pixels blank-pixels (+ height-scrollbar-1 height-blank-1))
 
-(defun company-box--change-line nil
+(defun company-box--point-at-line (&optional line start)
+  (save-excursion
+    (goto-char (or start 1))
+    (forward-line (or line company-selection 0))
+    (point)))
+
+(defun company-box--move-selection (&optional first-render)
   (let ((selection company-selection)
-        (common company-common))
+        (common company-common)
+        (candidates-length company-candidates-length)
+        (inhibit-redisplay t)
+        (inhibit-modification-hooks t))
+    ;;(message "MOVE-SELECTION WINDOW=%s" (get-buffer-window (company-box--get-buffer) t))
     (with-selected-window (get-buffer-window (company-box--get-buffer) t)
-      (company-box--update-line selection common))
-    (company-box--update-scrollbar (company-box--get-frame))))
-
-(defun company-box--next-line nil
-  (interactive)
-  (let ((max (min company-candidates-length company-box-max-candidates))
-        (next (1+ company-selection)))
-    (setq company-selection (if company-selection-wrap-around
-                                (mod next max)
-                              (min next (1- max))))
-    (company-box--change-line)
-    (company-box--update-width)))
-
-(defun company-box--prev-line nil
-  (interactive)
-  (let ((max (min company-candidates-length company-box-max-candidates))
-        (prev (1- company-selection)))
-    (setq company-selection (if (and company-selection-wrap-around (< prev 0))
-                                (1- max)
-                              (max prev 0)))
-    (company-box--change-line)
-    (company-box--update-width)))
-
-(defun company-box--start-changed-p nil
-  (not (equal company-box--start (window-start))))
-
-(defun company-box--post-command nil
-  (cond ((company-box--start-changed-p)
-         (company-box--on-start-change))))
+      (setq company-selection selection)
```

Comment:
Why don't you use `company-set-selection`?

Reply:
This statement is run in the company-box buffer, it doesn't interfere with the buffer the user is editing (the one where company-mode is running).
I might use another variable such as `company-box--parent-selection` but using `company-selection` avoid creating another variable, and `company-selection` is not used in this buffer